### PR TITLE
Single Types Feature Enhancement

### PR DESCRIPTION
**Pull request by meinvolk**
Allowing source plugin to accept Strapi Single Types. Have added a new array option in the gatsby-node for all single type items. Not sure if this is an 'ugly' solution or not. If so, please give me some advice on how to make it better. But, the solution does work! :)

### CODE REVIEW

**Comment by necheporenko** [comment]
Please check and merge this PR, it's a really hot feature.

**Comment by alexandrebodin** [comment]
Hi @meinvolk Can you please update your PR to support singleTypes as specific nodes with using a list for them ? It's a bit more work but would be a lot cleaner :)

**Comment by meinvolk** [comment]
@alexandrebodin

So the problem is when a Single Type is registered it will create a node in GraphQL that has "all" prepended on the type. For example, right now the plugin is creating allStrapiHomePage, and StrapiHomePage. We don't need allStrapiHomePage, since it is a single type and no collection is possible.

I have looked around the internet a ton for how to do this with no luck. I have emailed a Jason Lengstorf a Gatsby enthusiast and Netlify developer, and got the this response.

> To the best of my knowledge Gatsby doesn't support this by default, but you may want to reach out to Mikhail Novikov, who leads the GraphQL efforts at Gatsby — he'll know for sure and can give you insight in the how and why of things

I posted in the Gatsby Discord channel and got responses like. 
> "complicating it to make all not happen in certain cases seems wasteful and possibly more confusing than conforming to the general case." 

and 

> "in the two years of talking to gatsby users no one complained about that." 

What do you all think? I thought it was a good idea, but I am not sure if it is possible.

**Comment by daniellabonte** [comment]
Ugh yes 🔥🔥 I'm glad this is being addressed! I'm about to add this feature branch into my package if it's not released soon

File: src/fetch.js [hunk 1]
```diff
@@ -26,8 +28,11 @@ module.exports = async ({
   // Make API request.
   const documents = await axios(apiEndpoint, fetchRequestConfig)
 
+  // Make sure response is an array for single type instances
+  const response = documents.data.length ? documents.data : [ documents.data ]
```
Comment (alexandrebodin) [inline src/fetch.js]: use should rather use Array.isArray()

File: src/fetch.js [hunk 2]
```diff
@@ -36,7 +41,7 @@ module.exports = async ({
  * @param {object} item - Entry needing clean
  * @returns {object} output - Object cleaned
  */
-const clean = item => {
+const clean = (item, singleType) => {
```
Comment (alexandrebodin) [inline src/fetch.js]: you are not using the param

Comment (meinvolk) [reply src/fetch.js]: All set.

File: src/fetch.js (hunk 1 above)
Comment (meinvolk) [reply src/fetch.js]: All set.

**Comment by alexandrebodin** [comment]
Going to test today :)

File: src/fetch.js [hunk 3]
```diff
@@ -5,15 +5,17 @@ import pluralize from 'pluralize'
...
   jwtToken,
   queryLimit,
-  reporter,
+  reporter
 }) => {
   // Define API endpoint.
-  const apiBase = `${apiURL}/${pluralize(contentType)}`
+  let apiBase = singleType ? `${apiURL}/${singleType}` : `${apiURL}/${pluralize(contentType)}`
+  
   const apiEndpoint = `${apiBase}?_limit=${queryLimit}`
```
Comment (alexandrebodin) [inline src/fetch.js]: Can you remove the limit if you are not calling the fetch function with queryLimit (e.g singleType send a ?_limit=undefined)

Comment (meinvolk) [reply src/fetch.js]: The plugin uses a default of 100. I was not passing the query limit option from the gatsby-node into the singleTypes fetch data call. Both content types and single types are defaulting to 100 now with latest commit.

**Pull request merged**